openvdb/openvdb/tools/Merge.h

```diff
@@ -603,6 +645,9 @@ bool CsgUnionOrIntersectionOp<TreeT, Union>::operator()(RootT& root, size_t) con
             auto it = tiles.find(key);
             if (it != tiles.end() && it->second == INSIDE_STATE)     continue;
 
+            // Missing tiles are outside, so if we are intersection we skip
+            if (Intersect)  continue;
+
```

Comment:
This line successfully causes one direction of the intersection to work.  When the dest has fewer roots than a source, they aren't erroneously added.  But if you reverse the order the problem returns.